### PR TITLE
chore: Add changelog generator

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Part of https://github.com/apache/datafusion-comet/issues/394

## Rationale for this change

We need to be able to generate change logs as part of the release process.

## What changes are included in this PR?

- New Python script, based on the one that is used in DataFusion
- Updated README
- Example changelog for 0.1.0 (I can remove this before we merge the PR) - [rendered version](https://github.com/andygrove/datafusion-comet/blob/changelog-generator/dev/changelog/0.1.0.md)

### CODE REVIEW

File: dev/changelog/0.1.0.md
Comment (andygrove) [inline dev/changelog/0.1.0.md:20]: This is an example of the generated changelog. I added this to the PR to make it easier to review, but I plan on removing this before the PR is merged.

**Review by comphead — APPROVED** [review]
I'd say it is lgtm.

**Comment by andygrove** [comment]
Thanks for the review @comphead. I have removed the example changelog and will merge once CI is green.

**Pull request merged**
